File: csharp-netstandard/src/Cloud.Governance.Client/Model/PrincipalType.cs


using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.IO;
using System.Runtime.Serialization;
using System.Text;
using System.Text.RegularExpressions;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Linq;
using System.ComponentModel.DataAnnotations;
using OpenAPIDateConverter = Cloud.Governance.Client.Client.OpenAPIDateConverter;

namespace Cloud.Governance.Client.Model
{
    
    [JsonConverter(typeof(StringEnumConverter))]
    
    public enum PrincipalType
    {
        [EnumMember(Value = "0")]
        None = 0,

        [EnumMember(Value = "1")]
        User = 1,

        [EnumMember(Value = "2")]
        DistributionList = 2,

        [EnumMember(Value = "4")]
        Office365Group = 4,

        [EnumMember(Value = "8")]
        SharePointGroup = 8,

        [EnumMember(Value = "16")]
        Guest = 16,

        [EnumMember(Value = "32")]
        SecurityGroupOnly = 32,

        [EnumMember(Value = "64")]
        MailEnabledSecurityGroup = 64,

        [EnumMember(Value = "100")]
        SecurityGroup = 100,

        [EnumMember(Value = "111")]
        All = 111

    }

}

File: typescript/src/models/PrincipalType.ts
/* tslint:disable */
/* eslint-disable */
/*
 * Cloud Governance Api
 *
 * Contact: support@avepoint.com
 */

/**
 * 
 * @export
 * @enum {string}
 */
export enum PrincipalType {
    None = 0,
    User = 1,
    DistributionList = 2,
    Office365Group = 4,
    SharePointGroup = 8,
    Guest = 16,
    SecurityGroupOnly = 32,
    MailEnabledSecurityGroup = 64,
    SecurityGroup = 100,
    All = 111
}

export function PrincipalTypeFromJSON(json: any): PrincipalType {
    return PrincipalTypeFromJSONTyped(json, false);
}

export function PrincipalTypeFromJSONTyped(json: any, ignoreDiscriminator: boolean): PrincipalType {
    return json as PrincipalType;
}

export function PrincipalTypeToJSON(value?: PrincipalType | null): any {
    return value as any;
}

